### ISSUE TITLE
Implement TF.data in our code 

### ISSUE THREAD

**Issue by umang-sh** [thisisashukla/inclusive-ai#1]
https://www.tensorflow.org/guide/data

https://www.tensorflow.org/guide/data_performance

- [ ] .Implement TF.data in our code, using the current code and generators.

- [ ] Also follow optimisation steps

Acceptance criteria:

- [ ]  Fully functional inclusive AI training code with TF.Data API running and no manual data pulling

- [ ]  At least 1 Optimization step(s) followed and benchmarked

**Comment by dp1411** [comment]
Looking for implementation with respect to the read_anno function.

**Comment by Jagadeesh1111** [comment]
tf.data - Reducing memory footprint: A number of transformations, including interleave, prefetch, and shuffle, maintain an internal buffer of elements. If the user-defined function passed into the map transformation changes the size of the elements, then the ordering of the map transformation and the transformations that buffer elements affects the memory usage. In general, we recommend choosing the order that results in lower memory footprint, unless different ordering is desirable for performance. 

In the above para, **they recommend to choose the order of map transformation that results in lower memory foot print**

The tf.data API offers the tf.data.Dataset.map transformation, which applies a user-defined function to each element of the input dataset. Because input elements are independent of one another, the pre-processing can be parallelized across multiple CPU cores. 

**Map requires i/p elements to be independent which contradicts our sequential architecture**
 
And even google search didn't lead to any positive results. I think Samidh is right that tf.data might not help us to solve our Memory leak error.

**Comment by ShivamSGusain** [comment]
Facing conceptual issues in implementation due to difference in illustrated examples and our own model, particularly due to variation in the size of image tensor.
